sw/device/lib/crypto/api.h

```diff
@@ -0,0 +1,1378 @@
+  kMacModeKmac128 = 1,
+  // KMAC-256 mode.
+  kMacModeKmac256 = 2
+} mac_mode_t;
+
+/**
+ * Enum to define padding scheme for RSA data.
+ */
+typedef enum rsa_padding {
+  // Pads input data according to the PKCS#1-OAEP scheme.
+  kRsaPaddingOaep = 0,
+  // Pads input data according to the PKCS#1 (v1.5) scheme.
+  kRsaPaddingPkcs = 1,
+  // Pads input data according to the PKCS#1-PSS scheme.
+  kRsaPaddingPss = 2,
+  // NULL padding.
+  kRsaPaddingNull = 3
+} rsa_padding_t;
+
+/**
+ * Enum to define hash modes for RSA scheme
+ *
+ * Aligning with existing HASH modes.
+ */
+typedef enum rsa_hash {
+  // SHA2-256 hashing mode for RSA.
+  kRsaHashSha256 = 0,
+  // SHA2-384 hashing mode for RSA.
+  kRsaHashSha384 = 1,
+  // SHA2-512 hashing mode for RSA.
+  kRsaHashSha512 = 2,
+  // SHA3-384 hashing mode for RSA.
+  kRsaHashSha3_384 = 3
+} rsa_hash_t;
+
+/**
+ * Enum to define named elliptic curves.
+ */
+typedef enum ecc_named_curve {
+  // ECC named curve NIST P256.
+  kEccNamedCurveNistP256 = 0,
+  // ECC named curve NIST P384.
+  kEccNamedCurveNistP384 = 1,
+  // ECC named curve brainpool P256r1.
+  kEccNamedCurveBrainpoolP256R1 = 2
+} ecc_named_curve_t;
+
+/**
+ * Struct to handle ECDSA result.
+ */
+typedef struct ecc_signature {
+  // Length of ECC signature R parameter, in bytes.
+  size_t signature_r_len;
+  // ECC signature R parameter.
+  uint32_t *signature_r;
+  // Length of ECC signature S parameter, in bytes.
+  size_t signature_s_len;
+  // ECC signature S parameter.
+  uint32_t *signature_s;
+} ecc_signature_t;
+
+/**
+ * Enum to define EdDSA mode for signature.
+ */
+typedef enum eddsa_sign_mode {
+  // Signature mode EdDSA.
+  kEddsaSignModeEdDSA = 0,
+  // Signature mode Hashed EdDSA.
+  kEddsaSignModeHashEdDSA = 1
+} eddsa_sign_mode_t;
+
+/**
+ * Struct to handle ECC public key type.
+ *
+ * Length of public key coordinates (x,y) follows len(prime_P).
+ */
+typedef struct ecc_public_key {
+  // ECC public key x coordinate.
+  uint32_t *pubKey_x;
+  // ECC public key y coordinate.
+  uint32_t *pubKey_y;
+} ecc_public_key_t;
+
+/**
+ * Struct to handle EdDSA public key type.
+ *
+ * Length of encoded public key for ed25519 is 256 bits (32 bytes).
+ */
+typedef struct eddsa_public_key {
+  // Public key for ed25519
+  uint32_t *pubKey;
+} eddsa_public_key_t;
+
+/**
+ * Struct for domain parameters for a generic ECC Weierstrass curve.
+ */
+typedef struct ecc_domain {
+  // Length of prime P.
+  size_t len_prime_P;
+  // Value of prime P.
+  uint32_t *prime_P;
+  // Length of coefficient A.
+  size_t len_A;
+  // Value of coefficient A.
+  uint32_t *A;
+  // Length of coefficient B.
+  size_t len_B;
+  // Value of coefficient B.
+  uint32_t *B;
+  // Value of basepoint G’s x coordinate.
+  uint32_t *base_point_G_x;
+  // Value of basepoint G’s y coordinate.
+  uint32_t *base_point_G_y;
+  // Length of prime order of group.
+  size_t len_prime_order_Q;
+  // Value of prime order of group.
+  uint32_t *prime_order_Q;
```

Comment:
Some general thoughts on names...
- Lowercase everything.
- I think that `prime_P` and `base_point_G_x` can be simplified to `p` and `gx`, respectively.
- I'd replace `prime_order_Q` with just `group_order` or `order_of_Q`.
- I'd move `len` to be a prefix of variable names; that's what we do elsewhere in OT.